components/event_dispatcher/introduction.rst

```diff
@@ -502,84 +457,28 @@ event object. This means that all listeners have direct access to the
 object's :method:`Symfony\\Component\\EventDispatcher\\Event::getDispatcher`
 method.
 
-This can lead to some advanced applications of the ``EventDispatcher`` including
-letting listeners dispatch other events, event chaining or even lazy loading
-of more listeners into the dispatcher object. Examples follow:
+This can lead to some advanced applications of the ``EventDispatcher``
+including dispatching other events inside listeners, chaining events or even
+lazy loading listeners into the dispatcher object.
 
-Lazy loading listeners::
-
-    use Symfony\Component\EventDispatcher\Event;
-    use Acme\StoreBundle\Event\StoreSubscriber;
-
-    class Foo
-    {
-        private $started = false;
-
-        public function myLazyListener(Event $event)
-        {
-            if (false === $this->started) {
-                $subscriber = new StoreSubscriber();
-                $event->getDispatcher()->addSubscriber($subscriber);
-            }
-
-            $this->started = true;
-
-            // ... more code
-        }
-    }
-
-Dispatching another event from within a listener::
-
-    use Symfony\Component\EventDispatcher\Event;
-
-    class Foo
-    {
-        public function myFooListener(Event $event)
-        {
-            $event->getDispatcher()->dispatch('log', $event);
-
-            // ... more code
-        }
-    }
-
-While this above is sufficient for most uses, if your application uses multiple
-``EventDispatcher`` instances, you might need to specifically inject a known
-instance of the ``EventDispatcher`` into your listeners. This could be done
-using constructor or setter injection as follows:
-
-Constructor injection::
+While this is sufficient for most uses, if your application uses multiple
+dispatchers, you might need to specifically inject a known instance of the
+``EventDispatcher`` into your listeners::
```

Comment:
Honestly, I don't see the use case of injecting the event dispatcher into an event listener.